hal/src/peripherals/icm.rs

```diff
@@ -877,23 +910,19 @@ impl<I: RegionNum> Region<I> {
 pub struct Icm {
     /// ICM pac register providing hardware access
     icm: crate::pac::Icm,
+    _ahb_clk: IcmAhbClk,
+    _apb_clk: IcmApbClk,
 }
 
 impl Icm {
     /// Create the interface for the ICM peripheral
-    ///
-    /// Don't forget to enable the `APB` bus for ICM
-    ///
-    /// `AHB` bus is on by default at reset
-    ///
-    /// Clock::v1
-    /// `mclk.apbcmask.modify(|_, w| w.icm_().set_bit());`
-    ///
-    /// Clock::v2
-    /// `tokens.apbs.icm.enable();`
     #[inline]
-    pub fn new(icm: crate::pac::Icm) -> Self {
-        Self { icm }
+    pub fn new(icm: crate::pac::Icm, ahb_clk: IcmAhbClk, apb_clk: IcmApbClk) -> Self {
+        Self {
+            icm,
+            _ahb_clk: ahb_clk,
```

Comment:
With the underscore removed as above, clippy will want this to look like `ahb_clk,` r/t `ahb_clk: ahb_clk,`